### DISCUSSION TITLE
Is there a way to know if the internal cache was used?

### DISCUSSION THREAD

**Discussion by ryanm1-digital** [MichaCo/DnsClient.NET#191] [Q&A] [answered]
When UseCache is set (Default = True), is there a way to know if the DNSQueryResponse comes from the cache or a lookup has taken place? Thanks, Ryan.

https://dnsclient.michaco.net/docs/DnsClient.DnsQueryOptions.html

**Comment by MichaConrad** [comment, marked as answer]
The response currently does not have an indicator if the result was cached or not, no.
There will be a line item in the audit trail "log" though; and, if you attach a logger you should also get a debug log message.
